### PR TITLE
pass data object to generator + restore project creation from template panel bug fix 

### PR DESCRIPTION
**Pull request by slavik-lvovsky**

### CODE REVIEW

File: backend/src/extension.ts [hunk 1]
```diff
@@ -129,7 +129,7 @@ export class YeomanUIPanel {
 			vscodeYouiEvents, 
 			outputChannel, 
 			this.logger, 
-			{genFilter: this.genFilter, messages: this.messages, defaultNpmPaths},
+			{genFilter: this.genFilter, messages: this.messages, data: _.get(uiOptions, "data"), defaultNpmPaths},
```
Comment (tomer-epstein) [inline backend/src/extension.ts:132]: data: _.get(uiOptions, "data", {})

File: frontend/src/App.vue [hunk 2]
```diff
@@ -382,6 +382,9 @@ export default {
     async displayGeneratorsPrompt() {
       const uiOptions = await this.rpc.invoke("getState");
       this.messages = uiOptions.messages;
+      if (this.isInVsCode()) {
+        window.vscode.setState(uiOptions);
+      }
       await this.rpc.invoke("receiveIsWebviewReady", []);
```
Comment (tomer-epstein) [inline frontend/src/App.vue:388]: 1. rename func setMessages() to setState()
    async setState() {
      const uiOptions = await this.rpc.invoke("getState");
      this.messages = uiOptions.messages;
      if (this.isInVsCode()) {
        window.vscode.setState(uiOptions);
      }
    }
2. 
    async displayGeneratorsPrompt() {
      await this.setState();
      await this.rpc.invoke("receiveIsWebviewReady", []);
    },

File: backend/src/extension.ts (hunk 1 above)
Comment (slavik-lvovsky) [reply backend/src/extension.ts:132]: I do not want to create any object if it is not created by command. It is not our logic

File: frontend/src/App.vue (hunk 2 above)
Comment (slavik-lvovsky) [reply frontend/src/App.vue:400]: Done

File: backend/src/extension.ts [hunk 3]
```diff
@@ -129,7 +130,7 @@ export class YeomanUIPanel {
 			vscodeYouiEvents, 
 			outputChannel, 
 			this.logger, 
-			{genFilter: this.genFilter, messages: this.messages, defaultNpmPaths},
+			{genFilter: this.genFilter, messages: this.messages, data: _.get(uiOptions, "data", {}), defaultNpmPaths},
```
Comment (tomer-epstein) [inline backend/src/extension.ts:133]: can change:
this.genFilter = GeneratorFilter.create(_.get(uiOptions, "filter"));
to:
this.genFilter = GeneratorFilter.create(_.get(uiOptions, "genFilter"));

File: backend/src/extension.ts [hunk 4]
```diff
@@ -35,7 +35,8 @@ export function activate(context: vscode.ExtensionContext) {
 	context.subscriptions.push(vscode.commands.registerCommand("yeomanUI.toggleOutput", YeomanUIPanel.toggleOutput));
 
 	vscode.window.registerWebviewPanelSerializer(YeomanUIPanel.viewType, {
-		async deserializeWebviewPanel(webviewPanel: vscode.WebviewPanel, state: any) {
+		async deserializeWebviewPanel(webviewPanel: vscode.WebviewPanel, state?: any) {
+			_.set(state, "filter", _.get(state, "genFilter"));
```
Comment (tomer-epstein) [inline backend/src/extension.ts:39]: no need for -> _.set(state, "filter", _.get(state, "genFilter"));
already have genFilter on state obj.

Comment (slavik-lvovsky) [reply backend/src/extension.ts:39]: UI command passes only filter not genFilter
That is why when we try to rectreate UI we need to pass filter

File: backend/src/extension.ts (hunk 3 above)
Comment (slavik-lvovsky) [reply backend/src/extension.ts:133]: Same,
For the first time UI command passes only filter not genFilter

**Pull request merged**